### ISSUE TITLE
Onboarding screens

### ISSUE THREAD

**Issue by lukesleeman** [Western-Health-Covid19-Collaboration/wh_covid19_app#113]
We have the disclaimer which pops up when the app launches - but of course nobody really reads that.  There are a few key things it might be good to make really obvious to users when they first launch:

☑️ What is actually in the app: Info for aesthetic and critical care doctors and nurses.  PPE, intubation, etc
🙅‍♀️The app is informational.  Don't rely on it exclusively, confirm things against actual guidelines, use your judgment

Mike suggested some sort of onboarding pager thing they swipe through.

This may also be a good way to solve issue #108  - support for other hospitals.  On the last page we could ask them what hospital they work in.  For the time being it could be WH only, but the idea is we could add more to the list.

**Comment by workerbee22** [comment]
I think this is a good idea to give basic context of the app and lots of other first run information, and can integrate the current Disclaimer. I've done loads of this so I'm happy to pick it up, but we'll need to give a little thought to key messages we need to deliver in Onboarding?

**Comment by lukesleeman** [comment]
Priority on this is fairly low - i'll tag @greggmiller and label as medic, so we can get some thoughts on what needs to be in there.

**Comment by workerbee22** [comment]
👍Totally, but need to keep Onboarding in mind because the code is very intertwined, so don't want to fix Disclaimer, only to have to almost start again to add Onboarding. So going to just architect Disclaimer fix with Onboarding in mind, because we don't know the onbaording key messages yet.

**Comment by greggmiller** [comment]
Intention for the user (and WH perspective) would be to distil the key points of the disclaimer,  while reducing the harshness of the legal language and simply outline the apps purpose.

Key intentions would be:

1. the user feels that this app will be simple to use for purpose of staff welfare (PPE use) and critical care of the patient with COVID19. 

2. Every hospital has minor variations in their guidance, best to check with their own hospital their recommendations.

3. We try to keep it up to date, but that is also their responsibility.

4. The app suggests but does not prescribe, measures and actions that may be useful. It is the user's responsibility to make sure they are acting in the patients best interests. 

5. It is designed for Western Health anaesthetists and Crit. care doctors but others may use it as long as they are aware that the app is made available on the understanding that Western Health and its employees have no liability and neither do the designers, developers, testers, reviewers, organisers and all other contributors to the project for the user's actions.

(point 5 can particularly be condensed down).

I'm sure @workerbee22 @lukesleeman @Western-Health-Covid19-Collaboration/designers   will have a better ideas than me on how to do this 😄

**Comment by greggmiller** [comment]
One way to structure onboarding screens that would benefit the user, and safeguard those involved in developing the app, would be to breakdown the TestFlight information (full text provided at bottom).

_First screen_

This app is designed to serve as an educational tool and interactive cognitive aid for Western Health Anaesthetists and ICU doctors 👩‍⚕ 👨‍⚕ who are managing patients with COVID-19  🤒 .

🛑 ✋  It is ONLY for Clinician or hospital staff use  ✋ 🛑 

_Second screen_

We want to protect our staff from infection risk 🌡  and ensure excellent patient care 🏥 .
This app aims to provide a really quick, usable means to access the core information ℹ️ for this.

_3rd screen_
✋   However, it is not a comprehensive source nor can we guarantee it is completely up to date at the time of use 📱. 
It is created using Western Health guidelines, informally peer-reviewed and adapted, with permission, from College/Society guidelines.
🛑 It does not constitute official advice, it is your responsibility to ensure your practice is up to date, contextualised for the patient and in accordance with your institution's practice 🤓. 

_4th Screen_
We are keen for your suggestions on how this app can be improved 💬 .
Please fill the feedback form in the about section to do so 📈. 
This app runs in parallel with the Western Health COVID-19 Anaesthesia microsite https://whcovid19.wixsite.com/covid19
😄 

_Progress to home screen_
--------

Keen for thoughts @workerbee22 @lukesleeman @Western-Health-Covid19-Collaboration/designers 

I know we have this as low priority but depending on the work involved, it would be really useful to have in early builds as provides a lot of reassurance and breaks down the harshness of the legal language in Disclaimer.

-----------

TestFlight information:
"An initial version of the app, please install and make sure it starts correctly.

This app is designed to serve as an educational tool and an interactive cognitive aid for Western Health Anaesthetists and ICU doctors who are managing patients with COVID-19. 

The emphasis is on ensuring that we and our registrars get the most important aspects right for taking care of our own welfare (including PPE), our core skills (intubation and ventilation) and in any role we may have assisting our ICU colleagues.

The app provides a really quick, usable means to access cognitive aids required for these core skills.

We will be working to keep it up to date with any changes in our guidelines and we hope it’s imperfections are outweighed by its simplicity and functionality as a cognitive aid and educational tool.
This app does not replace the WH anaesthesia microsite but runs in parallel with it.
We are keen for suggestions on how it can be improved. 
Please fill the feedback form within the information section to do so. "

**Comment by workerbee22** [comment]
I just talked with Jenny from Apple on the phone with updates as follows:

- We can continue to submit Beta builds to TestFlight for the purposes of testing .... I'm finishing codemagic integration for this atm.
- But we will need to make the following changes PRIOR to submitting the app for final production Apple App Store review:

1. Provide a prominent Disclaimer .... which we are working on

2. Provide a prominent and clear message that the app is only for clinician/hospital staff use only (not the general public)

So this probably means finalising the Disclaimer and add Onboarding screens to explain the use/audience for the app as a high priority.

**Comment by lukesleeman** [comment]
I've assigned to version 2 - I think V1 will be just via TF anyway?

**Comment by marcedwards** [comment]
One potential issue with splitting this into pages: We have to ensure the longest page fits on the smallest device without scrolling (scrolling up and down + swiping left and right for pages = a bit confusing).

It seems like the original plan of just having a single scroll view, but with added headings could be good? I think it’s okay if this screen is pretty dry and uninteresting. It’s not supposed to look appealing to the general public.

**Comment by Mattro** [comment]
Unpacking this a bit, I feel the goals of the onboarding are:

**From Apple:**
- Provide a prominent Disclaimer .... which we are working on
- Provide a prominent and clear message that the app is only for clinician/hospital staff use only (not the general public)

Adding paging and stuff feels like a distraction, when we could lead the disclaimer with:

> This app is designed to serve as an educational tool and interactive cognitive aid for Western Health Anaesthetists and ICU doctors 👩‍⚕ 👨‍⚕ who are managing patients with COVID-19 🤒 .
> 
> 🛑 ✋ It is ONLY for Clinician or hospital staff use ✋ 🛑

Then the rest of that content Gregg has written up, and ship it.

**Comment by lukesleeman** [comment]
Oh, that sounds like a quick and pragmatic solution to solve the current problems we are facing 👍 

I think a good plan is to just put that in the disclaimer and then build out a set of proper onboarding screens later down the track once we have the rest of the app built out 🙂

I've updated issue #118 with the text

**Comment by workerbee22** [comment]
So we are all on the same page, onboarding something like this which is kind of standar on Android and iOS ???

- Pages that user swipes
- 1 and only 1 key message per page
- a graphic of some kind to visually reinforce the key message
- 'Get Start' button, Skip or other to exit onboarding and go to the app home page
- page dots (sorry bug the current page dot should be black

Something like this:

![onboarding example](https://user-images.githubusercontent.com/5154664/78110876-7608a700-7447-11ea-810d-202eb65688df.gif)

**Comment by Mattro** [comment]
Here's some Zeplin screens of how it can look.

https://app.zeplin.io/project/5e7597a68651056664fb6e53/dashboard?seid=5e84628a5dde8b4871f851da

**Comment by lukesleeman** [comment]
I'm removing design a medic from this until we get some time to re-visit it later

**Comment by workerbee22** [comment]
New conflicting requirements:
- @greggmiller has reported that management really wants (unlikely to budge) on the Disclaimer being the first screen which it is. But also want what is know currently as the 'Full Disclaimer' at the top.
- Apple requires 'prominent' guidance for users that the app audience and use is for hospital staff and not the general public.

With just one screen atm and management requiring the 'Full Disclaimer' at the top and above the content that describes the audience for the app, it is hard to argue that this guidance is 'prominent' when it is a long way down the screen.

So there is risk we get rejected by Apple on review, but this is uncertain. After much discussion with @lukesleeman and especially @greggmiller we have agreed to add Onboarding screens and convert the current disclaimer content to onboarding screens which is this currently:

![image](https://user-images.githubusercontent.com/5154664/81903842-b3934f00-9605-11ea-9202-8f2229ae2f70.png)
![image](https://user-images.githubusercontent.com/5154664/81903862-be4de400-9605-11ea-8f40-71a1d90e7fce.png)

This then reduces the Disclaimer back to the original content, current labelled as 'Full Disclaimer' which also scrolls much less.

@workerbee22 drafted some screens which seem acceptable but are just draft atm to confirm the idea:

![image](https://user-images.githubusercontent.com/5154664/81903991-f5bc9080-9605-11ea-90c6-0ecb96d9512a.png)

So tagging this as a Beta 4 milestone issue and moving up the priority to get done ASAP, and re present Beta 4 to management and move forward to final release of the app.

**Comment by workerbee22** [comment]
@greggmiller Approved the Onboarding screen PR 339 which is now merged. 

Closing.
